llvm/lib/DWARFLinker/DWARFStreamer.cpp

```diff
@@ -859,18 +859,20 @@ void DwarfStreamer::emitLineTablePrologueV5IncludeAndFileTable(
   for (auto Include : P.IncludeDirectories)
     emitLineTableString(P, Include, DebugStrPool, DebugLineStrPool);
 
-  bool InlineSources = any_of(P.FileNames, [](auto &File) {
-    auto s = dwarf::toString(File.Source);
-    return s && !**s;
-  });
+  bool HasChecksums = P.ContentTypes.HasMD5;
+  bool HasInlineSources =
+      P.ContentTypes.HasSource && any_of(P.FileNames, [](auto &File) {
```

Comment:
Oh! If `P.ContentTypes.HasSource` is a thing, we can probably delete the entire any_of expression?

Reply:
I wanted to double check the semantics, but yeah I think you're right. I was considering the optimization of actually checking that there's meaningful source code embedded there, but as I write it out I've pretty much convinced myself that's just making things complicated for barely/no gain.